### PR TITLE
fixing bug in wcssstr

### PR DESCRIPTION
**Pull request by mike-kaufman**
Fixing https://github.com/Microsoft/ChakraCore/issues/3434..  Bug in wcsstr would return a match if the string being searched ended with a prefix of the search string.  Fixed this to behave correctly.  Also added tests to test_native.sh to verify this.

This is re-opened from PR #3542.

### CODE REVIEW

File: test/native-tests/test-pal/sample.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    result += validate(PAL_wcsstr(u"targetstringxxx", u"searchStringxxx"), NULL, u"test1");
+    result += validate(PAL_wcsstr(u"abcdefg", u"def"), u"defg", u"test2");
+    result += validate(PAL_wcsstr(u"abcdefg", u"a"), u"abcdefg", u"test3");
+    result += validate(PAL_wcsstr(u"abcdefg", u"g"), u"g", u"test4");
+    result += validate(PAL_wcsstr(u"abcdefg", u"gh"), NULL, u"test5");
+    result += validate(PAL_wcsstr(u"abcdefg", u"ah"), NULL, u"test6");
+    result += validate(PAL_wcsstr((const char16_t *)NULL, (const char16_t *)NULL), NULL, u"test7");
+    result += validate(PAL_wcsstr((const char16_t *)NULL, u""), NULL, u"test8");
+    result += validate(PAL_wcsstr(u"", (const char16_t *)NULL), NULL, u"test9");
+    result += validate(PAL_wcsstr(u"abcdefg", u""), u"abcdefg", u"test10");
```
Comment (digitalinfinity) [inline test/native-tests/test-pal/sample.cpp]: These are good- i would also add cases for "abcdefg", "abcdefg", "", "" and "", "def"

Comment (mike-kaufman) [inline test/native-tests/test-pal/sample.cpp]: fixed

Comment (mike-kaufman) [inline test/native-tests/test-pal/sample.cpp]: the `"", ""` case actually uncovered another bug, where we would return `null` instead of the input string.  This was inconsistent with win32 API behavior, so I fixed that as well.

File: pal/src/cruntime/wchar.cpp [hunk 2]
```diff
@@ -1482,9 +1482,8 @@ PAL_wcsstr(
         goto leave;
     }
 
-    if (*strCharSet == 0)
-    {
-        ret = (char16_t *)string;
+    if (*strCharSet == 0) {
```
Comment (MSLaguana) [inline pal/src/cruntime/wchar.cpp]: This looks like an unintentional change, the curly brace should start on the line after the `if`

Comment (mike-kaufman) [inline pal/src/cruntime/wchar.cpp]: fixed...

File: test/native-tests/test-python/helloWorld.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
     platform = "so"
 
 build_type = sys.argv[1]
-if sys.argv[2] != None:
+if sys.argv.__len__() > 2 and sys.argv[2] != None:
```
Comment (MSLaguana) [inline test/native-tests/test-python/helloWorld.py]: Is this change required?

Comment (mike-kaufman) [inline test/native-tests/test-python/helloWorld.py]: I believe so.  I was getting failures when running the tests that used this about teh index being out of bounds.  Adding this checked made that error go away.
